File: backend-app/apiModule/ApiModule/WebApplication1/Domain/FormAnalysis.cs
namespace ApiModule.Domain;

public sealed class FormAnalysis
{
    public Guid Id { get; set; } = Guid.NewGuid();
    public Guid OwnerUserId { get; set; }

    public Guid? WorkoutRunId { get; set; }
    public WorkoutRun? WorkoutRun { get; set; }

    public Guid? WorkoutId { get; set; }
    public Workout? Workout { get; set; }

    public Guid? ExerciseId { get; set; }
    public string? ExerciseName { get; set; }
    public string ExerciseType { get; set; } = "squat";
    public int? StepIndex { get; set; }
    public int? SetNumber { get; set; }

    public string Status { get; set; } = "queued";
    public int? Score { get; set; }
    public string Summary { get; set; } = string.Empty;
    public string FindingsJson { get; set; } = "[]";
    public string MetricsJson { get; set; } = "[]";
    public string? RawResultJson { get; set; }
    public string? ErrorMessage { get; set; }

    public string SourceFileName { get; set; } = string.Empty;
    public string? AnalyzedFileName { get; set; }
    public string AnalyzerVersion { get; set; } = "form-analysis-v1";
    public string? ModelName { get; set; }

    public DateTime CreatedAt { get; set; }
    public DateTime? UpdatedAt { get; set; }
    public DateTime? CompletedAt { get; set; }
}

File: frontend-app/src/app/types/formAnalysis.ts
export type ExerciseFormAnalysisMetric = {
  label: string;
  value: string;
};

export type ExerciseFormAnalysisResult = {
  analysisId: string;
  workoutRunId?: string | null;
  workoutId?: string | null;
  exerciseId?: string | null;
  exerciseName?: string | null;
  exerciseType: string;
  stepIndex?: number | null;
  setNumber?: number | null;
  status: string;
  score?: number | null;
  summary: string;
  findings: string[];
  metrics: ExerciseFormAnalysisMetric[];
  hasSourceVideo: boolean;
  hasAnalyzedVideo: boolean;
  analyzerVersion?: string;
  modelName?: string | null;
  createdAt?: string;
  completedAt?: string | null;
};

export type ExerciseFormAnalysisKind = "source" | "analyzed";

export type ExerciseFormAnalysisUploadContext = {
  workoutRunId?: string;
  workoutId?: string;
  exerciseId?: string;
  exerciseName?: string;
  stepIndex?: number;
  setNumber?: number;
};

export type ExerciseFormAnalysisListFilters = {
  workoutRunId?: string;
  workoutId?: string;
};
